index.js

```diff
@@ -65,6 +65,14 @@ function formatWithCursor(text, opts, addAlignmentSize) {
   if (opts.requirePragma && !hasPragma(text)) {
     return { formatted: text };
   }
+  if (
+    opts.prependPragma &&
+    !hasPragma(text) &&
+    opts.rangeStart === 0 &&
+    opts.rangeEnd === Infinity
+  ) {
+    text = `/** @format */\n${text}`;
+  }
 
   text = stripBom(text);
```

Comment:
We probably want to do `stripBom` before we prepend the comment, not after.

Reply:
Thanks! Moved the `stripBom` in https://github.com/prettier/prettier/pull/2865/commits/a84df4567ba030a6e8cb5a2de85772590e8723a5